### ISSUE TITLE
Public chat/watching results

### ISSUE THREAD

**Issue by KitClement** [cubing/ccm#315]
Was watching results come in for One Night in Bejing on CubingChina's new system.  Really cool things to consider for CCM.

As scorecards are entered, the chat tells you what is getting entered so you can watch the results come in.  Makes it very interactive for those trying to keep up with the event.

![cubingchina](https://cloud.githubusercontent.com/assets/6166412/15632244/61b3962a-2540-11e6-85bc-d5396125b157.PNG)

As we move to an automated system rather than having data entry, this could be really cool -- imagine watching results come in as they happen and being able to be notified about what's happening at each competition!

(Just another note, if this does get implemented, would be good to have a way to hide results if organizers did not want others to see them yet -- for example, if finals are happening 2 at a time, it would be an unfair advantage for those who haven't competed yet to know the time they have to beat.)

**Comment by Baiqiang** [comment]
LOL. It's still under construction now. We just had a try on One Night in
Beijing 2016.

On 29 May 2016 at 16:56, KitClement <notifications@github.com> wrote:

> Was watching results come in for One Night in Bejing on CubingChina's new
> system. Really cool things to consider for CCM.
>
> As scorecards are entered, the chat tells you what is getting entered so
> you can watch the results come in. Makes it very interactive for those
> trying to keep up with the event.
>
> [image: cubingchina]
> <https://cloud.githubusercontent.com/assets/6166412/15632244/61b3962a-2540-11e6-85bc-d5396125b157.PNG>
>
> As we move to an automated system rather than having data entry, this
> could be really cool -- imagine watching results come in as they happen and
> being able to be notified about what's happening at each competition!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cubing/ccm/issues/315>, or mute the thread
> <https://github.com/notifications/unsubscribe/ACR5orxQoKdCZkgFWByukKC6eC2qE6oXks5qGVTagaJpZM4IpPu3>
> .
>



-- 
Baiqiang Dong
Website：http://www.qiyuuu.com

**Comment by jfly** [comment]
Awesome! Can you tell us more details about the new system, @Baiqiang?

Are you doing development somewhere on github?

**Comment by Baiqiang** [comment]
Yes. I developed a data entry system with chatting feature. As of now, every change of results will post a formatted message. All codes are in the develop branch of CubingChina: https://github.com/CubingChina/cubingchina/tree/develop.
There're some bugs to be fixed and lots of feature to be added. 
I used [Vue](http://vuejs.org/) for the frontend and [ratchet](https://github.com/ratchetphp/Ratchet/) to start up a websocket server. So I can make it more suitable to the current system.
